carbon-design-system/carbon-addons-iot-react
# fix: update useMerged hook to only render on prop changes

## Depends On: #3167 

**Summary**

- Fixes the error @stuckless pointed out here: https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3110#pullrequestreview-831972238
- [demo with using ...args, arguments, and a bad version](https://codesandbox.io/s/trusting-pateu-3n6xb?file=/src/Counter.js)

**Change List (commits, features, bugs, etc)**

- fixes useMerged 
- adds tests

**Acceptance Test (how to verify the PR)**

- confirm tests accurately cover the requirements
- confirm the demo using `useMergedArgs` works as expected.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- n/a



**Things to look for during review**
